### PR TITLE
Workaround for python3-debian bug about collecting control file

### PR DESCRIPTION
**Pull request by mbussolotto**
## What does this PR change?

Python-debian has changed the way it looks up the file names from the tar headers. Somehow they enforce the dot-slash (./) in the file name starting from the version 0.1.44, while it's totally possible to have a tar with file names as simple relative paths like control, not ./control. Since we cannot downgrade the version (introduced to fix https://github.com/SUSE/spacewalk/issues/17726 ) I just introduced a new method to do the same steps performed in version < 0.1.44  (as described here https://github.com/SUSE/spacewalk/issues/21530#issuecomment-1576992005 )

There's already a bug opened to fix issue in `python-debian` library https://www.mail-archive.com/pkg-python-debian-maint@alioth-lists.debian.net/msg00598.html

## GUI diff

No difference.

- [ ] **DONE**

## Documentation
- No documentation needed
- [x] **DONE**

## Test coverage
- https://ci.suse.de/view/Manager/view/Uyuni-PRs/job/uyuni-prs-ci-tests/2442/

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/21530
Fixes https://github.com/SUSE/spacewalk/issues/21212
Fixes https://github.com/SUSE/spacewalk/issues/20605
Fixes https://github.com/SUSE/spacewalk/issues/21699
Fixes https://github.com/uyuni-project/uyuni/issues/6396
Fixes https://github.com/SUSE/spacewalk/issues/21700
- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by mbussolotto** [comment]
acceptance tests: https://ci.suse.de/view/Manager/view/Uyuni-PRs/job/uyuni-prs-ci-tests/2436/

File: python/uyuni/common/rhn_deb.py [hunk 1]
```diff
@@ -34,6 +35,25 @@
...
+    # https://www.mail-archive.com/pkg-python-debian-maint@alioth-lists.debian.net/msg00598.html
+    # after the issue is fixed, remove this function
+    def get_file(self, control, fname):
+        if fname.startswith('./'):
+            fname = fname[2:]
+        elif fname.startswith('/'):
+            fname = fname[1:]
+
+        try:
+            fobj = control.tgz().extractfile(fname)
```
[the remark this answers is not in the document]
Comment (mbussolotto) [reply python/uyuni/common/rhn_deb.py:48]: Thanks for noticing it! I think it should be able to open also tar xz files https://salsa.debian.org/python-debian-team/python-debian/-/blob/master/lib/debian/debfile.py?ref_type=heads#L159 
```python
    def tgz(self):
        # type: () -> tarfile.TarFile
        """Return a TarFile object corresponding to this part of a .deb
        package.

        Despite the name, this method gives access to various kind of
        compressed tar archives, not only gzipped ones.
        """
```

 I'll try to sync some debian 11 deb and see what happens

[the remark this answers is not in the document]
Comment (vzhestkov) [reply python/uyuni/common/rhn_deb.py:48]: Maybe they just confusing with the method name. But it's better to double check for sure. Thanks for taking care of it :+1:

[the remark this answers is not in the document]
Comment (mbussolotto) [reply python/uyuni/common/rhn_deb.py:48]: Hi @vzhestkov , Is `control.tar.xz` in all Debian 11 packages ? I'm not sure how to check it, but I tried to sync https://deb.nodesource.com/node_18.x/dists/bullseye/main/binary-amd64/
 and the changes worked as expected.

[the remark this answers is not in the document]
Comment (mbussolotto) [reply python/uyuni/common/rhn_deb.py:48]: ok, I extracted the deb package and it has `control.tar.gz`..do you know some repo with packages with `control.tar.xz` ?

[the remark this answers is not in the document]
Comment (mbussolotto) [reply python/uyuni/common/rhn_deb.py:48]: tested with https://download.opensuse.org/repositories/systemsmanagement:/saltstack:/bundle:/Debian11/Debian_11/ and it works fine :)

[the remark this answers is not in the document]
Comment (vzhestkov) [reply python/uyuni/common/rhn_deb.py:48]: Thanks for checking :+1:

File: python/uyuni/common/rhn_deb.py [hunk 2]
```diff
@@ -34,6 +35,25 @@
...
+        except KeyError:
+            raise debfile.DebError("File not found inside package")
```
Comment (vzhestkov) [inline python/uyuni/common/rhn_deb.py:50]: Maybe explicitly specify that it refers to a control file?

**Review by vzhestkov — APPROVED** [review]
Just one more nitpicky suggestion, but other than that looks good for me.

Thanks for taking care of it :+1:

**Pull request merged**